inference-engine/include/ie_compound_blob.h

```diff
@@ -349,4 +349,64 @@ class INFERENCE_ENGINE_API_CLASS(I420Blob) : public CompoundBlob {
 
     Blob::Ptr createROI(const ROI& roi) const override;
 };
+
+/**
+ * @brief This class represents a blob that contains other blobs - one per batch
+ */
+class INFERENCE_ENGINE_API_CLASS(BatchedBlob) : public CompoundBlob {
+ public:
+    /**
+     * @brief A smart pointer to the BatchedBlob object
+     */
+    using Ptr = std::shared_ptr<BatchedBlob>;
+
+    /**
+     * @brief A smart pointer to the const BatchedBlob object
+     */
+    using CPtr = std::shared_ptr<const BatchedBlob>;
+
+    /**
+     * @brief Constructs a batched blob from a vector of blobs
+     * @details All passed blobs should meet following requirements:
+     * - all blobs have equal tensor descriptors,
+     * - blobs layouts should be one of: NCHW, NHWC, NCDHW, NDHWC, NC, CN, C, CHW
+     * - batch dimensions should be equal to 1 or not defined (C, CHW).
+     * Resulting blob's tensor descriptor is constructed using tensor descriptors
+     * of passed blobs by setting batch dimension to blobs.size()
+     *
+     * @param blobs A vector of blobs that is copied to this object
+     */
+    explicit BatchedBlob(std::vector<Blob::Ptr> blobs);
+
+    /**
+     * @brief Constructs a batched blob from a vector of blobs with specified
+     * tensor descriptor
+     * @details It is caller's responsibility to validate the correctness of blobs
+     * and tensor descriptor
+     *
+     * @param tensorDesc Defines the layout and dims of the blob
+     * @param blobs A vector of blobs that is copied to this object
+     */
+    BatchedBlob(std::vector<Blob::Ptr> blobs, const TensorDesc& tensorDesc);
+
+    /**
+     * @brief A virtual destructor
+     */
+    virtual ~BatchedBlob() = default;
```

Comment:
IMO, no need to define dtor and copy/move methods. Compiler will generate them by default.

Reply:
This definition is made just to follow the [rule-of-five](https://en.cppreference.com/w/cpp/language/rule_of_three) as far as copy/move .ctors and assignment operator had to be overridden.

Reply:
Just the note: copy/move operations had to be overridden due to different behavior than base class `CompoundBlob`

Reply:
What's about move assignment operator?

Reply:
If you look in the `CompoundBlob` [copy/move .ctors](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/inference_engine/ie_compound_blob.cpp#L199), those do not copy tensor descriptor, but `BatchedBlob` does.

Reply:
I would say it is a bug in `CompoundBlob`. @ilyachur @ilya-lavrenov  why do we need such logic in `CompoundBlob` ctor/move? Why can't we just use compiler default generated copy/move operations?

Reply:
Sorry, I am not so familiar with the big OV, to be sure if there is a bug or feature in `CompoundBlob` copy/move operations. And really do not know what is expected in the following case:
```cpp
NV12Blob a{y_blob_ptr, uv_blob_ptr};
CoumpondBlob b{a};
```

Reply:
@vinograd47 As I know we don't copy `TensorDesc` in the `CompoundBlob` because it is always entry.
In compound blob only sub-blobs have an initialised `TensorDesc`s.

Reply:
IMO, we should change this and use default copy/move operations (which copies `TensorDesc`). For some `CompoundBlob`s  common  `TensorDesc` can make sense (`BathedBlob`, `NV12Blob`).

Reply:
@vinograd47 , @ilyachur, 
IMHO, it would be nice to follow the 'rule-of-zero' everywhere.

Is the fix for `CompoundBlob` default operations to be done here or in a separate change?

Reply:
IMHO, it is relatively small change, so it can be done as a part of this PR.

Reply:
Agree to have defaults for `CompoundBlob` as its flavors where it's appropriate

Reply:
@rfsaliev are we going to finalize this?

Reply:
ilya-lavrenov, yes I am
TODO:
example code